### DISCUSSION TITLE
How should I classify an issue that suggests deprecating a feature?

### DISCUSSION THREAD

**Discussion by shahar1** [apache/airflow#31114] [Q&A] [answered]
I want to raise an issue that suggests deprecating a feature, but uncertain what kind of issue I should raise.
For that instance, "feature" could refer to one of the following use cases:
1. A deprecated parameter of 3rd party package being utilized in an operator/hook.
2. A parameter that is misused in its current form (for example: `self.num: int | str`, where the number should be used as an integer)
3. A parameter of a certain method that is redundant. For example: having `self.box: Box` and `self.boxes: list[Box]`, where I could use something like `self.boxes: Box | list[Box]`).
4. An operator/hook that becomes unnecessary for some reason.

As 1 could be classified as `Bug` (at least after upgrading the 3rd party package), the other cases are less clear: the software is not necessarily "buggy" in the sense that it doesn't work as expected, but implementing the deprecation does also exactly introduce a `New feature`.
Currently, in the ["New issue" menu](https://github.com/apache/airflow/issues/new/choose), `Task and meta` seems appropriate for this case - but it is classified as `(Maintainers only)`.
What should I do?

**Comment by potiuk** [comment, marked as answer]
Use whatever you think is appropriate and add description re-classifying it as needed. 
We can alwys re-classify it later.

At most you are risking that people will complain. Getting caught by missing option in processes where the processes have not foreseen the case you have is an indication of being bound to corporate red-tape - we are community and agree some rules and policies but if you think you need to act differently becasue current policies block you - just ACT.
